### ISSUE TITLE
`Style/NestedTernaryOperator` autocorrection raises when ternaries have an `if` statement between them

### ISSUE THREAD

**Issue by stevegeek** [rubocop/rubocop#12237]
If a nested ternary has a conditional between it and it's parent ternary, the cop raises `NoMethodError`

For example
```ruby
a ? (
  if b
    c ? 1 : 2
  end
) : 3
```

I have a fix ready, I will open a PR shortly

--------

## Expected behavior

The cop should autocorrect the example to

```ruby
if a
  if b
    c ? 1 : 2
  end
else
  3
end
```

## Actual behavior

The cop raises:

```
Failure/Error: replace_loc_and_whitespace(corrector, if_node.loc.question, "\n")
     
     NoMethodError:
       undefined method `question' for #<Parser::Source::Map::Condition:0x00007f3e94bb4430 @keyword=#<Parser::Source::Range (string) 8...10>, @end=#<Parser::Source::Range (string) 29...32>, @else=nil, @begin=nil, 
...
```

## Steps to reproduce the problem

The example provided above will cause the issue to happen when running rubocop and autocorrect.

## RuboCop version

```
$ rubocop -V
1.52.1 (using Parser 3.2.2.3, rubocop-ast 1.29.0, running on ruby 3.1.3) [arm64-darwin22]
```
